sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSub/src/Utilities.cs

```diff
@@ -141,5 +134,83 @@ public static RequestType GetRequestType(WebPubSubEventType eventType, string ev
+                if (signatures == null)
+                {
+                    continue;
+                }
+                using var hmac = new HMACSHA256(Encoding.UTF8.GetBytes(accessKey));
+                var hashBytes = hmac.ComputeHash(Encoding.UTF8.GetBytes(connectionId));
+                var hash = "sha256=" + BitConverter.ToString(hashBytes).Replace("-", "");
+                if (signatures.Contains(hash, StringComparer.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+                else
+                {
+                    continue;
+                }
+            }
+            return false;
+        }
+
+        public static bool ValidateMediaType(string mediaType, out MessageDataType dataType)
+        {
+            try
+            {
+                dataType = GetDataType(mediaType);
+                return true;
+            }
+            catch (Exception)
+            {
+                dataType = MessageDataType.Binary;
+                return false;
+            }
+        }
+
+        public static bool RespondToServiceAbuseCheck(HttpRequestMessage req, HashSet<string> allowedHosts, out HttpResponseMessage response)
+        {
+            response = new HttpResponseMessage();
+            if (req.Method == HttpMethod.Options || req.Method == HttpMethod.Get)
+            {
+                var requestHosts = req.Headers.GetValues(Constants.Headers.WebHookRequestOrigin);
+                return RespondToServiceAbuseCheck(requestHosts, allowedHosts, out response);
+            }
+            return false;
+        }
+
+        public static bool RespondToServiceAbuseCheck(HttpRequest req, HashSet<string> allowedHosts, out HttpResponseMessage response)
+        {
+            response = new HttpResponseMessage();
+            if (req.Method.Equals("options", StringComparison.OrdinalIgnoreCase) || req.Method.Equals("get", StringComparison.OrdinalIgnoreCase))
+            {
+                req.Headers.TryGetValue(Constants.Headers.WebHookRequestOrigin, out var requestHosts);
+                return RespondToServiceAbuseCheck(requestHosts, allowedHosts, out response);
+            }
+            return false;
+        }
+
+        private static bool RespondToServiceAbuseCheck(IEnumerable<string> requestHosts, HashSet<string> allowedHosts, out HttpResponseMessage response)
+        {
+            response = new HttpResponseMessage();
+            if (requestHosts != null && requestHosts.Any())
+            {
+                foreach (var item in allowedHosts)
+                {
+                    if (requestHosts.Contains(item))
+                    {
+                        response.Headers.Add(Constants.Headers.WebHookAllowedOrigin, requestHosts);
+                        return true;
+                    }
+                }
+                response.StatusCode = HttpStatusCode.BadRequest;
+            }
+            return true;
```

Comment:
>true [](http://example.com/codeflow?start=19&length=4)

always return true?

Reply:
Right. As methods are pre-checked, under this they're not upstream client requests and return true here to respond to requester quickly. So no need to parse request details anymore.